### PR TITLE
Activities in wiki

### PR DESCRIPTION
**Pull request by ajainuary**
**Description**
This PR refactors the code for creating new pages in the Wiki, for creating a LI in the Wiki and most importantly the `duplicateLI` function. Luckily, any old pages/wikis made previously would still work flawlessly.
**How to test**

1. Create an activity in the 'Individual Plane'
2. Use the activity from 2 different users (they will not share the data)

**Resolved Issues**
Asana task for reference: https://app.asana.com/0/1121331595459324/1125930882647548

### CODE REVIEW

File: frog/imports/api/generateReactiveFn.js [hunk 1]
```diff
@@ -187,7 +187,14 @@ export class Doc {
       'li',
       id
     );
-
+    if (LIData.liType === 'li-activity') {
+      // In this case the activity also needs to be duplicated
+      const acId = uuid();
+      const activityPointer = connection.get('rz', acId);
+      // New instance starts with empty data
+      activityPointer.create({});
```
Comment (houshuang) [inline frog/imports/api/generateReactiveFn.js:195]: this should be activityTypesObj[acId].dataStructure || {}

Comment (houshuang) [reply frog/imports/api/generateReactiveFn.js:195]: so the need for a specific duplication mechanism is exactly why we have the option of having a duplicate function in the liType... so can you move this logic to li-activity, adding a duplicate function to the main export?

Comment (houshuang) [reply frog/imports/api/generateReactiveFn.js:195]: also, should a new instance start with empty data even if there is already a lot of data? that's not my intuition of cloning - if I have a rich text with lots of text, and clone it, I get two lis with lots of text... cloning a ck-board should be the same?

File: frog/imports/client/Wiki/index.js [hunk 2]
```diff
@@ -466,45 +448,33 @@ class WikiComp extends Component<WikiCompPropsT, WikiCompStateT> {
...
 
-  createLI = (newTitle, plane, config, operatorConfig) => {
+  // Creates a new page entry in ShareDB and navigates to it.
+  createPage = (title, socialPlane, activityConfig, operatorConfig) => {
     const error =
-      checkNewPageTitle(wikiStore.parsedPages, newTitle) ||
-      (config.invalid && 'Activity config is not valid') ||
-      (operatorConfig.invalid && 'Operator config is not valid');
+      checkNewPageTitle(wikiStore.parsedPages, title) ||
+      (activityConfig &&
```
Comment (houshuang) [inline frog/imports/client/Wiki/index.js]: why not activityConfig?.invalid

**Review by houshuang — CHANGES_REQUESTED** [review]
Great start. See my comments on code - also when I try to create a new Individual Page with an activity type, the page gets created but the create modal stays open...

File: frog/imports/api/generateReactiveFn.js (hunk 1 above)
Comment (ajainuary) [reply frog/imports/api/generateReactiveFn.js]: Where should the instance get the new data from? (This is for creating a new instance when a new individual comes)

Comment (ajainuary) [reply frog/imports/api/generateReactiveFn.js:195]: If we export it into the `li-activity` we won't be able to assign the `id` of the activity to the `rz` field of the LI.

File: frog/imports/client/Wiki/index.js (hunk 2 above)
Comment (ajainuary) [reply frog/imports/client/Wiki/index.js]: Made the respective changes

**Review by houshuang — APPROVED** [review]

**Pull request merged**